src/Bicep.Core/TypeSystem/DeployTimeConstantDirectViolationVisitor.cs

```diff
@@ -117,14 +117,6 @@ private void FlagIfAccessingEntireResourceOrModule(SyntaxBase syntax)
                 }
             }
 
-            if (this.DeployTimeConstantContainer is not IfConditionSyntax and not ForSyntax)
-            {
-                // We can skip validation if we are inside a resource/module body or a function call, because the
-                // type checker should be able to produce type errors if the reference is invalid. There are also
-                // locations where referencing an entire resource/module module body is expected (e.g., scope).
-                return;
-            }
-
```

Comment:
This is no longer valid since 1) we changed type checking to only emit warnings for resources properties; 2) `identities`'s type is generic `Resource` will makes type checking useless.

But this makes me wonder if we should at least emit type errors for top level properties whose types cannot be customized by RPs, such as `location`.